src/test/java/org/broadinstitute/hellbender/utils/baq/BAQUnitTest.java

```diff
@@ -1,26 +1,230 @@
+        String readBases, refBases;
+        byte[] quals, expected;
+        String cigar;
+        int refOffset;
+        int pos;
+        ReferenceDataSource rds;
+
+        public BAQTest(String _refBases, String _readBases, String _quals, String _expected) {
+            this(_refBases, _readBases, _quals, _expected, null);
+        }
+
+        public BAQTest(String _refBases, String _readBases, String _quals, String _expected, ReferenceDataSource rds) {
+            this(0, -1, null, _readBases, _refBases, _quals, _expected, rds);
+        }
+
+        public BAQTest(int _refOffset, long _pos, String _cigar, String _refBases, String _readBases, String _quals, String _expected, ReferenceDataSource rds) {
+            refOffset = _refOffset;
+            pos = (int) _pos;
+            cigar = _cigar;
+            readBases = _readBases;
+            refBases = _refBases;
+            this.rds = rds;
+
+            quals = new byte[_quals.getBytes().length];
+            expected = new byte[_quals.getBytes().length];
+            for (int i = 0; i < quals.length; i++) {
+                quals[i] = (byte) (_quals.getBytes()[i] - 33);
+                expected[i] = (byte) (_expected.getBytes()[i] - 33);
+            }
+        }
+
+        public String toString() {
+            return readBases;
+        }
+
+        public GATKRead createRead() {
+            GATKRead read = ArtificialReadUtils.createArtificialRead(createHeader(), "foo", 0, pos > 0 ? pos + (refOffset > 0 ? refOffset : 0) : 1, readBases.getBytes(), quals);
+            read.setCigar(cigar == null ? String.format("%dM", quals.length) : cigar);
+            return read;
+        }
+    }
+
+
+    @DataProvider(name = "data")
+    public Object[][] createData1() {
+        List<BAQTest> params = new ArrayList<>();
+
+        SAMSequenceDictionary dict= new SAMSequenceDictionary();
+        dict.addSequence(new SAMSequenceRecord("1", Integer.MAX_VALUE));
+
+        params.add(new BAQTest(
+                "GCTGCTCCTGGTACTGCTGGATGAGGGCCTCGATGAAGCTAAGCTTTTTCTCCTGCTCCTGCGTGATCCGCTGCAG",
+                "GCTGCTCCTGGTACTGCTGGATGAGGGCCTCGATGAAGCTAAGCTTTTCCTCCTGCTCCTGCGTGATCCGCTGCAG",
+                "?BACCBDDDFFBCFFHHFIHFEIFHIGHHGHBFEIFGIIGEGIIHGGGIHHIIHIIHIIHGICCIGEII@IGIHCG",
+                "?BACCBDDDFFBCFFHHFIHFEIFHIGHHGHBFEIFGIIGEGII410..0HIIHIIHIIHGICCIGEII@IGIHCE"));
+
+        params.add(new BAQTest(
+                "GCTTTTTCTCCTCCTG",
+                "GCTTTTCCTCCTCCTG",
+                "IIHGGGIHHIIHHIIH",
+                "EI410..0HIIHHIIE"));
+
+        final String refString1 = "AAATTCAAGATTTCAAAGGCTCTTAACTGCTCAAGATAATTTTTTTTTTTTGAGACAGAGTCTTGCTGTGTTGCCCAGGCTGGAGTGCAGTGGCGTGATCTTGGCTCACTGCAAGCTCCGCCTCCCGGGTTCACGCCATTCTCCTGCCTCAGCCTCCCGAGTAGCTGGGACTACAGGCACCCACCACCACGCCTGGCCAATTTTTTTGTATTTTTAGTAGAGATAG";
+        final ReferenceDataSource rds1 = new ReferenceMemorySource(new ReferenceBases(refString1.getBytes(), new SimpleInterval("1", 9999807, 10000032)), dict);
```

Comment:
\[FindBugs\] WARNING: Dm: Found reliance on default encoding in org\.broadinstitute\.hellbender\.utils\.baq\.BAQUnitTest\.createData1\(\): String\.getBytes\(\)